pkg/kubectl-argo-rollouts/cmd/status/status.go

```diff
@@ -0,0 +1,123 @@
+package status
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/argoproj/argo-rollouts/pkg/kubectl-argo-rollouts/info"
+	"github.com/argoproj/argo-rollouts/pkg/kubectl-argo-rollouts/options"
+	"github.com/argoproj/argo-rollouts/pkg/kubectl-argo-rollouts/viewcontroller"
+	"github.com/spf13/cobra"
+)
+
+const (
+	statusLong = `Watch the progress of a rollout until it is done. The return will be success if
+the Rollout ends as Healthy and error if the Rollout ends degraded or if the
+timeout is hit before the progress finishes.`
+	statusExample = `
+	# Watch the rollout until it succeeds
+	%[1]s status gestbook
+
+	# Watch the rollout until it succeeds, fail if it takes more than 60 seconds
+	%[1]s status --timeout 60 guestbook
+	`
+)
+
+type StatusOptions struct {
+	Watch   bool
+	Timeout int64
+
+	options.ArgoRolloutsOptions
+}
+
+// NewCmdStatus returns a new instance of an `rollouts status` command
+func NewCmdStatus(o *options.ArgoRolloutsOptions) *cobra.Command {
+	statusOptions := StatusOptions{
+		ArgoRolloutsOptions: *o,
+	}
+
+	var cmd = &cobra.Command{
+		Use:          "status ROLLOUT_NAME",
+		Short:        "Show the status of a rollout.",
+		Long:         statusLong,
+		Example:      o.Example(statusExample),
+		SilenceUsage: true,
+		RunE: func(c *cobra.Command, args []string) error {
+			if len(args) != 1 {
+				return o.UsageErr(c)
+			}
+			name := args[0]
+			controller := viewcontroller.NewRolloutViewController(o.Namespace(), name, statusOptions.KubeClientset(), statusOptions.RolloutsClientset())
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+			controller.Start(ctx)
+
+			ri, err := controller.GetRolloutInfo()
+			if err != nil {
+				return err
+			}
+
+			if !statusOptions.Watch {
+				fmt.Fprintln(o.Out, ri.Status)
+			} else {
+				rolloutUpdates := make(chan *info.RolloutInfo)
+				defer close(rolloutUpdates)
+				controller.RegisterCallback(func(roInfo *info.RolloutInfo) {
+					rolloutUpdates <- roInfo
+				})
+				go statusOptions.WatchStatus(ctx.Done(), cancel, statusOptions.Timeout, rolloutUpdates)
+				controller.Run(ctx)
+
+				finalRi, err := controller.GetRolloutInfo()
+				if err != nil {
+					return err
+				}
+
+				if finalRi.Status == "Degraded" {
+					return fmt.Errorf("The rollout is in degraded state with message: %s", finalRi.Message)
+				} else if finalRi.Status != "Healthy" {
+					return fmt.Errorf("The rollout progress did not finish within the specified timeout")
```

Comment:
I don't understand `progress` in this context (see my previous request about the same)

Also, you can't use `timeout` like this. Timeout is a fixed quantity, whereas `within` operates on an interval.

Possibly something like:
```suggestion
					return fmt.Errorf("Timeout reached before rollout completed")
```

Reply:
Maybe using the same analogy as in `kubectl rollout status` might make sense here: https://github.com/kubernetes/kubectl/blob/e909d3918e23a879d24ca1fd624827037c9920a7/pkg/polymorphichelpers/rollout_status.go#L78

> deployment %q exceeded its progress deadline

Reply:
`exceeded` sounds good. Based on what you've written elsewhere, I don't think you can use `deadline`.

Reply:
The only thing I can think of is

> Rollout progress exceeded timeout

But I'm not sure if that is the best phrasing.